tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_multi_mount.ksh

```diff
@@ -0,0 +1,77 @@
+#!/bin/ksh -p
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy is of the CDDL is also available via the Internet
+# at http://www.illumos.org/license/CDDL.
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2018 Datto Inc
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify multi mount functionality using lazy unmount.
+#
+# STRATEGY:
+# 1. Create fs
+# 2. Create and hold open file in filesystem
+# 3. Lazy unmount
+# 4. Verify remount the fs that was lazily unmounted is possible
+# 5. Verify multiple mounts of the same dataset are possible
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	log_must pkill -P $TAILPPID tail
+	log_must umount $MNTFS2
+	log_must umount $MNTFS3
+	log_must zfs destroy $TESTFS
+	return 0
+}
+
+log_assert "Verify multiple mounts in one namespace are possible"
+
+log_onexit cleanup
+
+# 1. Create fs
+TESTFS="$TESTPOOL/multi-mount-test"
```

Comment:
Rather than redefine `TESTFS`, which is already defined in `default.cfg`, how about using the existing definitition and `$TESTPOOL/$TESTFS`.  Or alternately `TESTDS=$TESTPOOL/multi-mount-test` to avoid the reusing the variable name which could otherwise be confusing.